File: src/etch/bindings/csharp/support/AuthExceptionSerializer.cs
using System;
using System.Collections.Generic;
using System.Text;
using Etch.Msg;

namespace Etch.Support
{
    public class AuthExceptionSerializer : ImportExportHelper
    {
        private readonly static  String FIELD_NAME = "msg";
	
	    public static void Init( XType type, Class2TypeMap class2type )
	    {
		    Field field = type.GetField( FIELD_NAME );
		    class2type.Add(typeof(_Etch_AuthException), type );
		    type.SetComponentType( typeof(_Etch_AuthException));
		    type.SetImportExportHelper( new AuthExceptionSerializer( type, field ) );
		    type.PutValidator( field, Validator_string.Get( 0 ) );
		    type.Lock();
	    }
	
	    public AuthExceptionSerializer( XType type, Field field )
	    {
		    this.type = type;
		    this.field = field;
	    }
	
	    private readonly XType type;
	
	    private readonly Field field;
	
	    public StructValue exportValue( Object value )
	    {
		    StructValue sv = new StructValue( type );
		    sv.Add( field, value.ToString() );
		    return sv;
	    }

	    public Object importValue( StructValue sv )
	    {
		    sv.CheckType( type );
		    return new _Etch_AuthException( (String) sv.Get( field ) );
	    }

    }
}

File: src/etch/bindings/java/support/AuthExceptionSerializer.java
/* $Id$
 *
 * Created by sccomer on Apr 18, 2008.
 *
 * Copyright (c) 2005 Metreos, Inc. All rights reserved.
 */

package etch.bindings.java.support;

import etch.bindings.java.msg.Field;
import etch.bindings.java.msg.ImportExportHelper;
import etch.bindings.java.msg.StructValue;
import etch.bindings.java.msg.Type;

public class AuthExceptionSerializer implements ImportExportHelper
{
	private final static String FIELD_NAME = "msg";
	
	public static void init( Type type, Class2TypeMap class2type )
	{
		Field field = type.getField( FIELD_NAME );
		class2type.put( _Etch_AuthException.class, type );
		type.setComponentType( _Etch_AuthException.class );
		type.setImportExportHelper( new AuthExceptionSerializer( type, field ) );
		type.putValidator( field, Validator_string.get( 0 ) );
		type.lock();
	}
	
	public AuthExceptionSerializer( Type type, Field field )
	{
		this.type = type;
		this.field = field;
	}
	
	private final Type type;
	
	private final Field field;
	
	public StructValue exportValue( Object value )
	{
		StructValue sv = new StructValue( type );
		sv.put( field, value.toString() );
		return sv;
	}

	public Object importValue( StructValue struct )
	{
		struct.checkType( type );
		return new _Etch_AuthException( (String) struct.get( field ) );
	}
}
